tools/install-git-hooks.js

```diff
@@ -0,0 +1,99 @@
+import files from './files';
+
+const hookDestination =
+  files.pathJoin(files.getCurrentToolsDir(), '.git', 'hooks');
+
+const hookSource =
+  files.pathJoin(files.getCurrentToolsDir(),
+    'scripts', 'admin', 'git-hooks');
+
+const allPossibleHooks = [
+  'applypatch-msg',
+  'pre-applypatch',
+  'post-applypatch',
+  'pre-commit',
+  'prepare-commit-msg',
+  'commit-msg',
+  'post-commit',
+  'pre-rebase',
+  'post-checkout',
+  'post-merge',
+  'pre-receive',
+  'update',
+  'post-update',
+  'pre-auto-gc',
+  'post-rewrite'
+];
+
+// A suffix that we append to our hook scripts
+const METEOR_HOOK_SUFFIX = '.meteor-hook';
+
+// Matches the whole line we inject into the user's hook
+const METEOR_HOOK_LINE_REGEX = /.*meteor-hook.*/;
+
+export default function installGitHooks() {
+  allPossibleHooks.map((hookName) => {
+    const hookFile = files.pathJoin(hookDestination, hookName);
+    const sourceFile = files.pathJoin(hookSource, hookName);
+    const meteorHookFile = hookFile + METEOR_HOOK_SUFFIX;
+
+    if (!files.exists(sourceFile) && !files.exists(hookFile)) {
+      // Don't do anything if this hook isn't in either list
+      return;
+    }
+
+    if (!files.exists(sourceFile)) {
+      // If the user has this hook, but Meteor doesn't, make sure that we remove
+      // any hooks that we added previously
+      removeMeteorInjectedHook(hookFile);
+      return;
+    }
+
+    /* eslint-disable max-len */
+    const METEOR_HOOK_INJECTED_LINE = `exec "$(dirname $0)/${hookName}${METEOR_HOOK_SUFFIX}" # Inserted by Meteor tool\n`;
+    /* eslint-enable */
+
+    if (!files.exists(hookFile)) {
+      // If the user doesn't have this hook, but we want to add it, just insert
+      // a new file and a line that calls it
+      files.writeFile(hookFile,
+        `#!/bin/sh
+${METEOR_HOOK_INJECTED_LINE}`,
+        { encoding: 'utf8', mode: 0o777 });
+
+      // Copy the hook from the scripts dir
+      files.writeFile(meteorHookFile, files.readFile(sourceFile),
+        { mode: 0o777 });
```

Comment:
That's a great point, I'm just really bad at unix permissions. I should make it only executable by the current user.